### PR TITLE
Render shopping lists and the items in them

### LINKED ISSUE #2: 2. As a user, I want to read all of my shopping lists and the items in them.
## Summary

The most important feature of our app is the ability to look at their shopping lists to see what's on them, so the user knows what they need to buy. For this to work, we'll have to be able to choose a our shopping lists. The `List` (the active shopping list) and `Home` (where the user will be able to switch shopping lists) components are the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `useShoppingLists` hook is used in `App.jsx` to get the shopping lists a user has access to from the database
- [x] The lists are passed into the `Home` component as a prop named `data`
- [x] The `setListPath` function is passed into the `Home` component so we can update the current list
- [x] The `useShoppingListData` hook is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [ ] In `Home.jsx`, the `SingleList` component is used to render the name of each shopping list a user is subscribed to
- [ ] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` or `SingleList.jsx` to show more information – you will do that **later**.
- The variables you need are already being passed into the `List` and `Home` components. What steps do you need to take in order to render them?
- Additionally, until issues [4](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/issues/4) and [5](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/issues/5) you should only be able to see the shopping list that was manually added in [issue 1](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/issues/1).

### PR DESCRIPTION
**Pull request by trushmi**
## Description
Made changes to Home component to display the name of each shopping list a user is subscribed to. Refactored the List component to display all of the items of the selected list. 

## Related Issue

Closes #2.

## Acceptance Criteria

- [X]  In Home.jsx, the SingleList component is used to render the name of each shopping list a user is subscribed to

- [X]  In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes

rendering components 

### Before
Screenshot 2024-02-06 at 2 37 34 PM
Screenshot 2024-02-06 at 2 37 17 PM

### After

Screenshot 2024-02-06 at 2 23 44 PM

Screenshot 2024-02-06 at 2 23 27 PM

## Testing Steps / QA Criteria

git pull
git switch it-pc-feature-list-rendering
Run npm ci
Run npm start
Navigate to home page to see a list 
Navigate to /list route to see a list of items

### CODE REVIEW

**Review by jaguilar89 — APPROVED** [review]
Looks good guys! I am able to navigate to /lists and see the items rendered out on the page.

image

**Review by rachelspencer — COMMENTED** [review]
Looks great!

**Review by rachelspencer — APPROVED** [review]

File: src/views/List.jsx [hunk 1]
```diff
@@ -7,11 +7,9 @@ export function List({ data }) {
 				Hello from the <code>/list</code> page!
 			</p>
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `data` array
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
+				{data.map((item, index) => (
+					<ListItem name={item.name} key={index} />
```
Comment (belindablack) [inline src/views/List.jsx:11]: Suggestion: As the key, you could use the `item.id` instead of the index for more specificity, since it's available.

**Review by belindablack — APPROVED** [review]
Pulled in the changes to test and the Acceptance Criteria was met. Great job!

**Pull request merged**